docs/common/CHARM_LINKING.md

```diff
@@ -0,0 +1,447 @@
+
+### Step 1: Define Separate Input and Output Interfaces
+
+```typescript
+/// <cts-enable />
+import { Cell, Default, NAME, recipe, UI, lift } from "commontools";
+
+// Data structure to expose
+interface Stats {
+  average: number;
+  q1: number;
+  median: number;
+  q3: number;
+  iqr: number;
+  count: number;
+}
+
+// Input: what this charm accepts
+interface Input {
+  name: Default<string, "my-stats-v1">;
+  rawData: Default<string, "">;
+}
+
+// Output: what this charm exposes for linking
+interface Output {
+  name: string;
+  rawData: string;
+  computedStats: Stats | null;  // <-- Exposed for other charms
+}
+```
+
+### Step 2: Compute and Return the Exposed Data
+
+```typescript
+const calculateStats = lift((values: number[]): Stats | null => {
+  if (values.length === 0) return null;
+
+  const sorted = [...values].sort((a, b) => a - b);
+  const n = sorted.length;
+  const sum = sorted.reduce((acc, val) => acc + val, 0);
+  const average = sum / n;
+
+  // Quartile calculations...
+  const q1 = percentile(sorted, 25);
+  const median = percentile(sorted, 50);
+  const q3 = percentile(sorted, 75);
+  const iqr = q3 - q1;
+
+  return { average, q1, median, q3, iqr, count: n };
+});
+
+export default recipe<Input, Output>(
+  "Stats Calculator",
+  ({ name, rawData }) => {
+    // Compute derived values
+    const parsedValues = parseData(rawData);
+    const computedStats = calculateStats(parsedValues);
+
+    return {
+      [NAME]: "Stats Calculator",
+      [UI]: (/* your UI here */),
+
+      // These fields are exposed for linking:
+      name,
+      rawData,
+      computedStats,  // <-- Other charms can link to this
+    };
+  }
+);
+```
+
+### Key Rules for Source Charms
+
+1. **Every field in the return object is linkable** - include what you want to expose
+2. **Use `lift()` for computed values** - ensures reactivity
+3. **Type the Output interface explicitly** - documents your charm's API
+4. **Nested fields are accessible** - `charmId/computedStats/average` works
+
+---
+
+## Part 2: Creating a Consumer Charm (Data Receiver)
+
+A consumer charm declares expected linked data using `Default<T, null>` in its Input interface.
+
+### Step 1: Define Input with Optional Linked Fields
+
+```typescript
+/// <cts-enable />
+import { Default, NAME, recipe, UI, lift } from "commontools";
+
+// Must match the source charm's exposed structure
+interface Stats {
+  average: number;
+  q1: number;
+  median: number;
+  q3: number;
+  iqr: number;
+  count: number;
+}
+
+// Input declares optional linked fields with Default<T, null>
+interface Input {
+  name: Default<string, "stats-reader-v1">;
+  linkedStats: Default<Stats | null, null>;  // null until linked
+}
+```
+
+### Step 2: Handle Both Linked and Unlinked States
+
+```typescript
+// Helper functions handle null gracefully
+const hasData = lift((s: Stats | null) => s !== null);
+const getAverage = lift((s: Stats | null) => s?.average);
```

Comment:
getAverage that just reads a property is not necessary, you can just write formatNumber(linkedStats.average) below in the JSX